### ISSUE TITLE
Infinite loop on small images

### ISSUE THREAD

**Issue by roed** [amakaseev/sprite-sheet-packer#66]
Mac user here. It seems that when I use images smaller than 10px (10px width, 9px height, png), the application hangs and seems to go in an infinite loop. It says: Optimizing atlas in the bottom right corner. I have to force quit it to stop it. When I use an image of 10x10px, it's fine. I searched in the source code, but I am having a hard time to find out why it does that. There is a for and while loop after that statement that displays the message:
```
bool SpriteAtlas::packWithRect(const QVector<PackContent>& content) {
```

**Comment by amakaseev** [comment]
Hi, please send me examples of files which cause this mistake, and the file of the project with settings

**Comment by roed** [comment]
![spider](https://user-images.githubusercontent.com/944060/59911683-c2cd4500-9414-11e9-8818-eaec5bfac309.png)

json of the project file:
```
{
    "algorithm": "Rect",
    "dataFormat": "pixijs",
    "destPath": "../../assets",
    "encryptionKey": "",
    "epsilon": 15,
    "heuristicMask": false,
    "imageFormat": "*.png",
    "jpgQuality": 80,
    "pixelFormat": "ARGB8888",
    "pngOptLevel": 7,
    "pngOptMode": "None",
    "premultiplied": true,
    "prependSmartFolderName": true,
    "rotateSprites": false,
    "scalingVariants": [
        {
            "forceSquared": false,
            "maxTextureSize": 2048,
            "name": "",
            "pow2": false,
            "scale": 1
        }
    ],
    "spriteBorder": 0,
    "spriteSheetName": "spritesheet",
    "srcList": [
    ],
    "textureBorder": 0,
    "trimMode": "Rect",
    "trimSpriteNames": true,
    "trimThreshold": 0,
    "webpQuality": 80
}
```

**Comment by amakaseev** [comment]
I used this sprites: 
![11](https://user-images.githubusercontent.com/9881001/59914926-23b04980-9424-11e9-9acd-a311884a4ba8.png)
![22](https://user-images.githubusercontent.com/9881001/59914927-23b04980-9424-11e9-949e-5387ea8f75d8.png)
![33](https://user-images.githubusercontent.com/9881001/59914928-2448e000-9424-11e9-8524-b711edb55161.png)

and sprite sheet worked correctly

**Comment by roed** [comment]
did you also try the sprite I provided?

**Comment by amakaseev** [comment]
i found mistake
int step = (w + h) / 20;

incorrect calculate steps
try fix this: int step = max( (w + h) / 20, 1);

**Comment by roed** [comment]
ah, so It would resolve to 0, hanging somewhere without doing any steps?

**Comment by amakaseev** [comment]
yes, see last [commit](https://github.com/amakaseev/sprite-sheet-packer/commit/6f84441aab02e9bcd28ae6f4c611de27d2b24d79)
Thanks for the help :)

**Comment by roed** [comment]
Nice. Btw, the latest release is from 2017, will you create a new release soon?

**Comment by amakaseev** [comment]
Oh I did not do it so long ago:)

**Comment by roed** [comment]
37 commits behind master ;)

**Comment by jjhesk** [comment]
crashed when i choose a smaller size of the sprite sheet
